satpy/composites/__init__.py

```diff
@@ -585,43 +588,99 @@ def __call__(self, projectables, **kwargs):
             LOG.debug("Computing sun zenith angles.")
             # Get chunking that matches the data
             try:
-                chunks = day_data.sel(bands=day_data['bands'][0]).chunks
+                chunks = data_1.sel(bands=data_1['bands'][0]).chunks
```

Comment:
Github won't let me comment on it, but the `coszen = ` line above uses `projectables[2]` which may or may not exist depending on the day/night mode.

Reply:
This `projectables[2]` remains in the original one. I actually don't know what it's for...May be it's waiting for a solar zenith angle band/dataset from the satellite data?

Reply:
:facepalm: Oh duh. That is if the YAML file specifies the solar_zenith_angle dataset from the reader (ex. for `viirs_sdr`). Geostationary satellites don't typically have that provided by the file/reader so it isn't specified. You may want this to be `projectables[2 if self.day_night == "day_night" else 1]`.

Reply:
😊OK It's late night here so I'll do these changes tomorrow.